pkg/stream/http2/stream.go

```diff
@@ -802,6 +803,8 @@ func (s *clientStream) AppendHeaders(ctx context.Context, headersIn api.HeaderMa
 	var req *http.Request
 	var isReqHeader bool
 
+	headersIn.Del(protocol.MosnHeaderScheme)
```

Comment:
`headersIn.Del(protocol.MosnHeaderScheme)` should be called after `headersIn = headersIn.Clone()`.

Reply:
nice catch! fixed